contracts/receivables/BaseCreditPoolReceivable.sol

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import {BaseStructs as BS} from "../libraries/BaseStructs.sol";
+import "../BaseCreditPool.sol";
+
+/**
+ * @title BaseCreditPoolReceivable
+ * @dev Contract that represents an ERC721 receivable payable to Huma's BaseCreditPool
+ */
+contract BaseCreditPoolReceivable is ERC721Enumerable, ERC721URIStorage, AccessControl {
+    using Counters for Counters.Counter;
+    using BS for BS.CreditRecord;
+
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    Counters.Counter private _tokenIdCounter;
+
+    struct ReceivableInfo {
+        // The BaseCreditPool that's paid by the receivable
+        BaseCreditPool baseCreditPool;
```

Comment:
Do not think we should include pool as an attribute for a receivable. When the pool has tons of receivables, this will lead to lots of redundant storage. You will have a collection (mapping) of the receivables, will it make more sense to have the pool address as the mapping key?

Reply:
As discussed, let us keep poolAddress so that we can use subgraph to get the list of receivables for a pool.